lib/phoenix/live_dashboard/page_builder.ex

```diff
@@ -230,394 +224,445 @@ defmodule Phoenix.LiveDashboard.PageBuilder do
+            <div>
+              <%= usage.title %>
+              <.hint text={usage[:hint]} :if={usage[:hint]}/>
+            </div>
+            <div>
+              <small class="text-muted pr-2">
+                <%= usage.current %> / <%= usage.limit %>
+              </small>
+              <strong :if={usage[:percent]}>
+                <%= usage[:percent] %>%
+              </strong>
+            </div>
+          </.title_bar_component>
+        <% end %>
+      </div>
+    </div>
+    """
+  end
 
-    {UsageCardComponent, assigns}
+  @doc false
+  attr :class, :string, default: ""
+  attr :color, :string, default: "blue"
+  attr :dom_id, :string, required: true
+  attr :percent, :float, required: true
+  attr :csp_nonces, :any, required: true
+  slot(:inner_block, required: true)
+
+  defp title_bar_component(assigns) do
+    ~H"""
+    <div class={@class}>
+      <section>
+        <div class="d-flex justify-content-between">
+          <%= render_slot @inner_block %>
+        </div>
+        <style nonce={@csp_nonces.style}>#<%= "#{@dom_id}-progress" %>{width:<%= @percent %>%}</style>
+        <div class="progress flex-grow-1 mt-2">
+          <div
+          class={"progress-bar bg-#{@color}"}
+          role="progressbar"
+          aria-valuenow={@percent}
+          aria-valuemin="0"
+          aria-valuemax="100"
+          id={"#{@dom_id}-progress"}
+          >
+          </div>
+        </div>
+      </section>
+    </div>
+    """
   end
 
   @doc """
-  Renders a shared usage card component.
-
-  It can be rendered in any dashboard page via the `render_page/1` function:
-
-      def render_page(assigns) do
-        shared_usage_card(
-          usages: [
-            %{
-              data: [
-                {"Atoms", 1.4, "green", nil},
-                {"Binary", 9.1, "blue", nil},
-                {"Code", 31.5, "purple", nil},
-                {"ETS", 3.6, "yellow", nil},
-                {"Processes", 25.8, "orange", nil},
-                {"Other", 28.5, "dark-gray", nil}
-              ],
-              dom_sub_id: "total"
-            }
-          ],
-          dom_id: "memory",
-          total_data: [
-            {"Atoms", 737513, "green", nil},
-            {"Binary", 4646392, "blue", nil},
-            {"Code", 16060819, "purple", nil},
-            {"ETS", 1845584, "yellow", nil},
-            {"Processes", 13146728, "orange", nil},
-            {"Other", 14559276, "dark-gray", nil}
-          ],
-          total_legend: "Total usage:"
-          total_usage: "47.4 MB"
-        )
-      end
+  Shared usage card component.
 
   You can see it in use the Home page and OS Data pages.
-
-  # Options
-
-  These are the options supported by the component:
-
-    * `:usages` - Required. A list of `Map` with the following keys:
-      * `:data` - A list of tuples with 4 elements with the following data:
-        `{usage_name, usage_percent, color, hint}`
+  """
+  attr :usages, :list,
```

Comment:
I wonder if we should make this a slot? If we slots we can do:

```
<:usage data={...} dom_sub_id={...} title={...} />
```

Or would that make the usage harder?

Reply:
It makes sense to use slots when we expect a list. It is nicer to read.